legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-logical-plan/src/main/java/org/finos/legend/engine/persistence/components/planner/BitemporalDeltaPlanner.java

```diff
@@ -113,12 +104,21 @@ class BitemporalDeltaPlanner extends BitemporalPlanner
     {
         super(datasets, ingestMode, plannerOptions);
 
+        if (ingestMode().validityMilestoning().validityDerivation() instanceof SourceSpecifiesFromDateTime && ingestMode().deduplicationStrategy() instanceof FilterDuplicates)
```

Comment:
Why do we check the instance of Validity Derivation?

Reply:
The use of this extra staging table only exists in SourceSpecifiesFromDateTime mode